OMCompiler/Compiler/NFFrontEnd/NFInstNode.mo

```diff
@@ -1101,6 +1107,41 @@ uniontype InstNode
     end match;
   end scopeListClass;
 
+  function getAnnotation
+    input String name;
+    input InstNode node;
+    output Option<SCode.SubMod> mod = NONE();
+  algorithm
+    mod := match node
+      local
+        Pointer<Component> component;
+        list<SCode.SubMod> subModLst;
+        Boolean done = false;
+
+      case COMPONENT_NODE(component = component)
+      then match Pointer.access(component)
+        case Component.TYPED_COMPONENT(
+          comment = SOME(SCode.COMMENT(
+          annotation_ = SOME(SCode.ANNOTATION(
+          modification = SCode.MOD(subModLst = subModLst))))))
+        algorithm
```

Comment:
This would be better written as something like:
```Modelica
if InstNode.isComponent(node) then
   mod := match Component.comment(InstNode.component(node))
     case SOME(SCode.ANNOTATION(modification = SCode.MOD(subModLst = subModLst)))
```